pennylane/collections/dot.py

```diff
@@ -121,17 +129,32 @@ def dot(x, y):
 
         interface = x.interface
         fn, _ = _get_dot_func(interface)
-        func = lambda params, **kwargs: fn(x(params, **kwargs), y(params, **kwargs))
+
+        if interface == "jax":
+            import jax.numpy as jnp
+            func = lambda params, **kwargs: fn(jnp.array(x(params, **kwargs)), jnp.array(y(params, **kwargs)))
+        else:
+            func = lambda params, **kwargs: fn(x(params, **kwargs), y(params, **kwargs))
```

Comment:
@AlaricCheng, how come you need to branch here specifically for the JAX interface?

If `x` and `y` are both QNodes with the JAX interface, than they should always be outputting JAX arrays, shouldn't they?

Reply:
Hi Josh,

The outputs are actually lists of JAX arrays. For example,
```python
dev = qml.device("default.qubit", wires = 2)

@qml.qnode(dev, interface = "jax")
def x_rotations(params):
    qml.RX(params[0], wires=0)
    qml.RX(params[1], wires=1)
    qml.CNOT(wires=[0, 1])
    return qml.expval(qml.PauliZ(0))

@qml.qnode(dev, interface = "jax")
def y_rotations(params):
    qml.RY(params[0], wires=0)
    qml.RY(params[1], wires=1)
    qml.CNOT(wires=[0, 1])
    return qml.expval(qml.Hadamard(0))

qnodes = qml.QNodeCollection([x_rotations, y_rotations])
print(qnodes([0.2, 0.1]))
```
The output would be 
```python
[DeviceArray(0.9800666, dtype=float32), DeviceArray(0.7070364, dtype=float32)]
```
That's why I put an extra branch for JAX (I know it looks annoying :)). Otherwise, it will raise an error.

Reply:
Got it! Thanks for clarifying @AlaricCheng! I think I know why it has this behaviour: in `QNodeCollection.convert_results`, you need to add a branch there to `jnp.stack(results)`. This will convert it into a `DeviceArray` rather than a _list_ of DeviceArrays, and will allow you to remove the branches here 🙂

Reply:
It works. I now remove the branches.